vignettes/lab_grading.Rmd

```diff
@@ -0,0 +1,395 @@
+  "AST", "AST", "Aspartate Aminotransferase (U/L)", 5,
+  "BASO", "BASO", "Basophils (10^9/L)", 6,
+  "BASOLE", "BASOLE", "Basophils/Leukocytes (FRACTION)", 7,
+  "BILI", "BILI", "Bilirubin (umol/L)", 8,
+  "BUN", "BUN", "Blood Urea Nitrogen (mmol/L)", 9,
+  "CA", "CA", "Calcium (mmol/L)", 10,
+  "CHOL", "CHOLES", "Cholesterol (mmol/L)", 11,
+  "CK", "CK", "Creatinine Kinase (U/L)", 12,
+  "CL", "CL", "Chloride (mmol/L)", 13,
+  "COLOR", "COLOR", "Color", 14,
+  "CREAT", "CREAT", "Creatinine (umol/L)", 15,
+  "EOS", "EOS", "Eosinophils (10^9/L)", 16,
+  "EOSLE", "EOSLE", "Eosinophils/Leukocytes (FRACTION)", 17,
+  "GGT", "GGT", "Gamma Glutamyl Transferase (U/L)", 18,
+  "GLUC", "GLUC", "Glucose (mmol/L)", 19,
+  "HBA1C", "HBA1C", "Hemoglobin A1C (1)", 20,
+  "HCT", "HCT", "Hematocrit (1)", 21,
+  "HGB", "HGB", "Hemoglobin (mmol/L)", 22,
+  "K", "POTAS", "Potassium (mmol/L)", 23,
+  "KETONES", "KETON", "Ketones", 24,
+  "LYM", "LYMPH", "Lymphocytes (10^9/L)", 25,
+  "LYMLE", "LYMPHLE", "Lymphocytes/Leukocytes (FRACTION)", 26,
+  "MACROCY", "MACROC", "Macrocytes", 27,
+  "MCH", "MCH", "Ery. Mean Corpuscular Hemoglobin (fmol(Fe))", 28,
+  "MCHC", "MCHC", "Ery. Mean Corpuscular HGB Concentration (mmol/L)", 29,
+  "MCV", "MCV", "Ery. Mean Corpuscular Volume (f/L)", 30,
+  "MICROCY", "MICROC", "Microcytes", 31,
+  "MONO", "MONO", "Monocytes (10^9/L)", 32,
+  "MONOLE", "MONOLE", "Monocytes/Leukocytes (FRACTION)", 33,
+  "PH", "PH", "pH", 34,
+  "PHOS", "PHOS", "Phosphate (mmol/L)", 35,
+  "PLAT", "PLAT", "Platelet (10^9/L)", 36,
+  "POIKILO", "POIKIL", "Poikilocytes", 37,
+  "POLYCHR", "POLYCH", "Polychromasia", 38,
+  "PROT", "PROT", "Protein (g/L)", 39,
+  "RBC", "RBC", "Erythrocytes (TI/L)", 40,
+  "SODIUM", "SODIUM", "Sodium (mmol/L)", 41,
+  "SPGRAV", "SPGRAV", "Specific Gravity", 42,
+  "TSH", "TSH", "Thyrotropin (mU/L)", 43,
+  "URATE", "URATE", "Urate (umol/L)", 44,
+  "UROBIL", "UROBIL", "Urobilinogen", 45,
+  "VITB12", "VITB12", "Vitamin B12 (pmol/L)", 46,
+  "WBC", "WBC", "Leukocytes (10^9/L)", 47
+)
+
+adlb <- lb %>%
+  ## Add PARAMCD PARAM and PARAMN - from LOOK-UP table
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = vars(PARAMCD, PARAM, PARAMN),
+    by_vars = vars(LBTESTCD)
+  ) %>%
+  ## Calculate PARCAT1 AVAL AVALC ANRLO ANRHI
+  ## Dummy the values for BASE
+  mutate(
+    PARCAT1 = LBCAT,
+    AVAL = LBSTRESN,
+    AVALC = LBSTRESC,
+    ANRLO = LBSTNRLO,
+    ANRHI = LBSTNRHI,
+    BASE = AVAL - 10
+  )
+```
+
+Another look-up table is used to add on `ATOXDSCL` and `ATOXDSCH` using
+`PARAMCD`. `ATOXDSCL` holds the terms for grading low lab values, and `ATOXDSCH` holds 
+the terms for grading high lab values. The names of these variables can be user-defined.
+
+```{r, eval=TRUE}
+# Assign ATOXDSCL and ATOXDSCH to hold lab grading terms
+# ATOXDSCL and ATOXDSCH hold terms defined by NCI-CTCAEv4.
+grade_lookup <- tibble::tribble(
+  ~PARAMCD, ~ATOXDSCL,                      ~ATOXDSCH,
```

Comment:
Maybe specify the point of this lookup table is to link the PARAMCD with the grading criteria metadata? I think it can help avoid confusion from users why we need to create those two variables.

Reply:
added some more text, let me know if its sufficient.